### PR TITLE
clientv3: do not refresh token when users use CommonName based authentication

### PR DESCRIPTION
**Pull request by ahrtr**
Fix https://github.com/etcd-io/etcd/issues/14764

Signed-off-by: Benjamin Wang <wachao@vmware.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by ahrtr** [comment]
When users use CommonName based authentication, then no need to fresh token at all. And actually the [authTokenBundle](https://github.com/etcd-io/etcd/blob/8438ddae3e738c1ed0c92f67b511741a08ce1edd/client/v3/client.go#L70) is always nil, so the client will crash when refreshing the token. 

We do not have any e2e test cases to cover the Common Name based authentication so far. So I added the first one in this PR.

@mitake  Please take a look at this PR. thx. cc @ptabor @serathius @spzala

**Comment by ahrtr** [comment]
I just noticed that we already have 3 cases to cover the CommonName based authentication in [ctl_v3_auth_no_proxy_test.go](https://github.com/etcd-io/etcd/blob/main/tests/e2e/ctl_v3_auth_no_proxy_test.go), it's good news.

File: tests/e2e/ctl_v3_auth_no_proxy_test.go [hunk 1]
```diff
@@ -34,3 +39,103 @@ func TestCtlV3AuthCertCNAndUsername(t *testing.T) {
...
+func TestCtlV3AuthCertCNWithWithConcurrentOperation(t *testing.T) {
+	e2e.BeforeTest(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// apply the certificate which has `root` CommonName,
+	// and reset the setting when the test case finishes.
+	t.Log("Apply certificate with root CommonName")
+	resetCert := applyTLSWithRootCommonName()
+	defer resetCert()
```
Comment (serathius) [inline tests/e2e/ctl_v3_auth_no_proxy_test.go:52]: There should be a separate cert bundle and an option in e2e tests pick certs with CommonName.

Comment (ahrtr) [reply tests/e2e/ctl_v3_auth_no_proxy_test.go:52]: I thought about enhancing the e2e framework, but it needs more time. I will think about it in a separate PR. Let's get 3.5.6 out firstly.

Comment (ahrtr) [reply tests/e2e/ctl_v3_auth_no_proxy_test.go]: Just rebased this PR, and added a TODO comment. PTAL.

Comment (serathius) [reply tests/e2e/ctl_v3_auth_no_proxy_test.go]: Sounds good.

**Review by serathius — APPROVED** [review]

**Comment by mitake** [comment]
@ahrtr Thanks a lot for fixing the issue, the change LGTM

**Comment by ahrtr** [comment]
> @ahrtr Thanks a lot for fixing the issue, the change LGTM

Thanks @mitake for the confirmation.

**Pull request merged**